### PR TITLE
WRP-12255: Modify description of minimum node version requirement

### PR DESCRIPTION
**Pull request by sjro**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Documents related to docs need to be updated for minimum node version requirements. 
Because docs has its dependency updated to docs-utils 0.4.x with documentationjs 14.0.0 or higher

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Update CHANGLOG.md, README.md and package.json to announce the minimum node version.
Deleted caveats as it builds on windows as well.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRP-12255

### Comments
Enact-DCO-1.0-Signed-off-by: SJ RO (sj.ro@lge.com)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "description": "Enact JavaScript Framework Docs",
   "version": "4.0.1",
   "engines": {
-    "node": ">=8.0.0"
+    "node": ">=14"
```
Comment (seunghoh) [inline package.json:6]: Do we fix all issues over node 17?

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Change Log
```
Comment (seunghoh) [inline CHANGELOG.md:1]: Is this the first time to make a change log in the docs repo?

Comment (sjro) [reply CHANGELOG.md:1]: Yes.
A tag version was created, but there was no change log, so it is difficult to remember history and leave a log.
Added CHANGELOG.md file to leave logs in the future.

File: package.json (hunk 1 above)
Comment (sjro) [reply package.json:6]: Yes.
Currently, travis check in the docs develop branch passes without problems on nodes 18 and 19.
https://github.com/enactjs/docs/pull/207/checks

**Review by seunghoh — APPROVED** [review]
LGTM

**Pull request merged**